addons/mass_mailing_sms/models/sms_sms.py

```diff
@@ -22,9 +22,9 @@ def _update_body_short_links(self):
                 continue
 
             body = sms.body
-            for url in re.findall(tools.TEXT_URL_REGEX, body):
+            for url in set(re.findall(tools.TEXT_URL_REGEX, body)):
```

Comment:
Note that we do a bit the same kind of processing with emails in ``_send_prepare_body`` in mass_mailing/models/mail_mail.py, did you have a look ? Notably if we have several identical emails (also doing a re.findall).